ocaml/xapi/xapi_host.ml

```diff
@@ -1340,10 +1340,34 @@ let get_uncooperative_resident_VMs ~__context ~self = []
 (* Dummy implementation for a deprecated API method. *)
 let get_uncooperative_domains ~__context ~self = []
 
+let replace_host_certificate ~__context ~host ~complete_task
+    (write_cert_fs : unit -> X509.Certificate.t) : unit =
+  (* NOT atomic
```

Comment:
Do we want a lock of some type to make it atomic?

Reply:
With this comment I meant that if we write a cert to the file system but the database writes fail then we'll get into a dodgy state. However now that I removed the code which does `pool_uninstall` this is probably less of a concern (but could still happen)

Nonetheless a lock would be good in case two api users try to modify the certs at the same time